giskard/ml_worker/websocket/listener.py

```diff
@@ -806,3 +806,26 @@ def get_push_objects(client: Optional[GiskardClient], params: websocket.GetPushP
     }
 
     return push_functions[params.push_kind](model, dataset, df)
+
+
+@websocket_actor(MLWorkerAction.createSubDataset)
+def create_sub_dataset(
+    client: Optional[GiskardClient], params: websocket.CreateSubDatasetParam, *arg, **kwargs
+) -> websocket.CreateSubDataset:
+    dataset = Dataset.download(
+        client=client,
+        project_key=params.projectKey,
+        dataset_id=params.dataset.id,
+        sample=params.dataset.sample
+    )
+
+    sub_dataset = Dataset(
+        df=dataset.df.iloc[params.rowIndexes],
```

Comment:
we should have `loc` here. `loc` selects rows using row labels whereas `iloc` selects rows using their integer positions, in the tests we define typically the failed indices like that:
```py
failed_idx = results_df.loc[~results_df.index.isin(passed_idx)].index.values
```
so we should then have:
```py
df.loc[failed_idx]
```

Reply:
As pointed out by @Inokinoki , the indices we send to the hub are expected to be integers. So we should return positional indices of the failed examples from tests from now on: https://linear.app/giskard/issue/GSK-2140/return-integer-positional-indices-for-failed-samples-in-testresult

Reply:
Actually the `failed_idx = results_df.loc[~results_df.index.isin(passed_idx)].index.values` is only used for the test logic.

Afterward we are sending `list(dataset.df.loc[failed_idx].index)` to the backend to save the list of failled tests. So I guess we should keep `iloc`